traffic_monitor/datareq/crstate_test.go

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package datareq
+
+import (
+	"github.com/apache/trafficcontrol/traffic_monitor/towrap"
+	"sync"
+	"testing"
+
+	"github.com/apache/trafficcontrol/lib/go-tc"
+	"github.com/apache/trafficcontrol/traffic_monitor/peer"
+	"github.com/apache/trafficcontrol/traffic_monitor/todata"
+)
+
+func TestUpdateStatusAnycast(t *testing.T) {
+
+	localStates := peer.NewCRStatesThreadsafe()
+	sameIpServers := map[tc.CacheName]map[tc.CacheName]bool{}
+
+	toData := todata.NewThreadsafe()
+	towrap.TrafficOpsSessionThreadsafe{}
+	toData.Update()
+
+	TODataThreadsafe{m: &sync.RWMutex{}, toData: New()}
+
+	localStates.AddCache("available", tc.IsAvailable{IsAvailable: true, Ipv4Available: true, Ipv6Available: true})
+	localStates.AddCache("tooHigh", tc.IsAvailable{IsAvailable: false, Ipv4Available: true, Ipv6Available: true, Status: "too high"})
```

Comment:
Maybe you can reuse (or copy) this code that does something similar?

https://github.com/apache/trafficcontrol/blob/d95a16c57f69cc186da237919be61387bf154e95/traffic_monitor/ds/stat_test.go#L59-L79